airflow/decorators/base.py

```diff
@@ -382,7 +382,15 @@ def expand(self, **map_kwargs: OperatorExpandArgument) -> XComArg:
         prevent_duplicates(self.kwargs, map_kwargs, fail_reason="mapping already partial")
         # Since the input is already checked at parse time, we can set strict
         # to False to skip the checks on execution.
-        return self._expand(DictOfListsExpandInput(map_kwargs), strict=False)
+        if self.is_teardown:
+            if "trigger_rule" in self.kwargs:
+                raise ValueError("Trigger rule not configurable for teardown tasks.")
+            self.kwargs.update(trigger_rule=TriggerRule.ALL_DONE_SETUP_SUCCESS)
```

Comment:
This should probably be in `__attrs_post_init__`? Also covers the non-mapped case.